### PR TITLE
Support for nested sub-menus on OS X

### PR DESCRIPTION
**Pull request by oxtoacart**
This simplifies the code from #99 and fixes the problem with sub-menu items getting duplicated on update.

### CODE REVIEW

File: systray.go [hunk 1]
```diff
@@ -109,6 +129,16 @@ func AddSeparator() {
 	addSeparator(atomic.AddInt32(&currentID, 1))
 }
 
+// AddSubMenuItem adds nested sub-menu item with designated title and tooltip, returning a channel
+// that notifies whenever that menu item is clicked.
```
Comment (joesis) [inline systray.go]: Not new in this PR, but both this and `AddMenuItem` now return a struct rather than a channel.

File: systray_darwin.m [hunk 2]
```diff
@@ -156,16 +190,15 @@ - (void) hide_menu_item:(NSNumber*) menuId
...
 
-- (void)setMenuItemIcon:(NSArray*)imageAndMenuId {
+- (void) setMenuItemIcon:(NSArray*)imageAndMenuId {
   NSImage* image = [imageAndMenuId objectAtIndex:0];
   NSNumber* menuId = [imageAndMenuId objectAtIndex:1];
 
   NSMenuItem* menuItem;
-  int existedMenuIndex = [menu indexOfItemWithRepresentedObject: menuId];
-  if (existedMenuIndex == -1) {
+  menuItem = find_menu_item(menu, menuId);
```
Comment (joesis) [inline systray_darwin.m]: I guess other occurrences of `indexOfItemWithRepresentedObject`, i.e., `show_menu_item`/`hide_menu_item`, need to be replaced by `find_menu_item` as well.

Comment (oxtoacart) [reply systray_darwin.m]: Good catch! Will fix.

File: systray.go (hunk 1 above)
Comment (oxtoacart) [reply systray.go]: Fixed.

File: systray_darwin.m (hunk 2 above)
Comment (oxtoacart) [reply systray_darwin.m]: Fixed.

**Comment by joesis** [comment]
Thanks for tackling this out @oxtoacart !

**Comment by joesis** [comment]
And @tstromberg @Arshiamidos your previous work !

**Pull request merged**